entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/UpdateConverter.java

```diff
@@ -89,31 +79,39 @@ public class UpdateConverter implements Converter<AttributeUpdateOperation, SubD
   public SubDocumentUpdate convert(
       final AttributeUpdateOperation operation, final RequestContext context)
       throws ConversionException {
-    final AttributeOperationMetadata metadata = OPERATOR_MAP.get(operation.getOperator());
+    final UpdateOperator operator = OPERATOR_MAP.get(operation.getOperator());
 
-    if (metadata == null) {
+    if (operator == null) {
       throw new ConversionException(
           String.format("Operator %s is not supported yet", operation.getOperator()));
     }
 
     final String id = operation.getAttribute().getColumnName();
-    final Optional<String> pathOptional =
-        entityAttributeMapping.getDocStorePathByAttributeId(context, id);
+    final Optional<AttributeKind> attributeKindOptional =
+        entityAttributeMapping.getAttributeKind(context, id);
 
-    if (pathOptional.isEmpty()) {
+    if (attributeKindOptional.isEmpty()) {
       throw new ConversionException(String.format("Cannot update non-attribute value for %s", id));
     }
 
-    final String subDocPath = pathOptional.orElseThrow();
-    final UpdateOperator operator = metadata.operator();
+    final AttributeKind attributeKind = attributeKindOptional.orElseThrow();
```

Comment:
nit - duplicative of 93. Instead remove that if condition, 

```suggestion
    final AttributeKind attributeKind = attributeKindOptional.orElseThrow(() => some error);
```